File: examples/code/triangle-cpp_bug/triangle.cpp
#include "triangle.hpp"

int classify_triangle(double a, double b, double c) {
  double tmp;

  // Sort the sides so that a <= b <= c
  if(a > b) {
    tmp = a;
    a = b;
    b = tmp;
  }

  if(a > c) {
    tmp = a;
    a = c;
    c = tmp;
  }

  if(b > c) {
    tmp = b;
    b = c;
    c = tmp;
  }

  if(a + b <= c)
    return INVALID;
  if(a == b && b == c)
    // should be EQUILATERAL
    return ISOSCELES;
  if(a == b || b == c)
    // should be ISOSCELES
    return EQUILATERAL;
  return SCALENE;
}

File: examples/triangle-rb/triangle.rb
def classify_triangle(a, b, c)
  # sort the sides so that a <= b <= c
  if a > b
    tmp = a
    a = b
    b = tmp
  end

  if a > c
    tmp = c
    a = c
    c = tmp
  end

  if b > c
    tmp = b
    b = c
    c = tmp
  end

  case
  when a + b <= c
    return :invalid
  when a == b && b == c
    return :equilateral
  when a == b || b == c
    return :isosceles
  else
    return :scalene
  end
end
